ci/setup_env.sh

```diff
@@ -0,0 +1,135 @@
+#!/bin/bash -e
+
+
+# edit the locale file if needed
+if [ -n "$LOCALE_OVERRIDE" ]; then
+    echo "Adding locale to the first line of pandas/__init__.py"
+    rm -f pandas/__init__.pyc
+    SEDC="3iimport locale\nlocale.setlocale(locale.LC_ALL, '$LOCALE_OVERRIDE')\n"
+    sed -i "$SEDC" pandas/__init__.py
+    echo "[head -4 pandas/__init__.py]"
+    head -4 pandas/__init__.py
+    echo
+    sudo locale-gen "$LOCALE_OVERRIDE"
+fi
+
+MINICONDA_DIR="$HOME/miniconda3"
+
+
+if [ -d "$MINICONDA_DIR" ]; then
+    echo
+    echo "rm -rf "$MINICONDA_DIR""
+    rm -rf "$MINICONDA_DIR"
+fi
+
+echo "Install Miniconda"
+UNAME_OS=$(uname)
+if [[ "$UNAME_OS" == 'Linux' ]]; then
+    if [[ "$BITS32" == "yes" ]]; then
+        CONDA_OS="Linux-x86"
+    else
+        CONDA_OS="Linux-x86_64"
+    fi
+elif [[ "$UNAME_OS" == 'Darwin' ]]; then
+    CONDA_OS="MacOSX-x86_64"
+else
```

Comment:
so how does this work on windows?

Reply:
Windows can't use this script unfortunately, as it's a shell script. In Windows, we have `ci/incremental/setup_conda_environment.cmd` for setting up the environment, and conda is downloaded directly in the azure configuration with powershell commands.